test/libsolidity/SemanticTest.cpp

```diff
@@ -88,19 +103,29 @@ SemanticTest::SemanticTest(string const& _filename, langutil::EVMVersion _evmVer
 
 TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePrefix, bool _formatted)
 {
+	set<pair<bool, bool>> viaYul;
```

Comment:
This is getting really complicated. Can we extract the actual test run to a function that gets booleans about what it should compile? Then we maybe don't even have to run this in a weird loop over booleans but can just call this function multiple times in several lines?

Reply:
Yes, I totally agree. I introduced a private method `runTest`, that is basically exactly the same as the original `run` method.  The current `run` method may now call `runTest` multiple times, it just depends whether `m_runWithoutYul`, `m_runWithYul || m_enforceViaYul` or `m_runWithEwasm` was set. I think it is much easier to read now.